### PR TITLE
⚡ aws: make ElastiCache kmsKey resolution lazy to speed up cacheClusters

### LINKED ISSUE #5115: slow aws query
```
        "query": "aws.elasticache.clusters.all(_['SnapshotRetentionLimit'] \u003e _['SnapshotRetentionPeriod'])"
```

this query has been reported to be slow, we should investigate how to improve performance

### PR DESCRIPTION
**Pull request by tas50**
## Problem

Issue #5115 reports a slow `aws.elasticache` query that only filters on snapshot retention:

```
aws.elasticache.cacheClusters.all(snapshotRetentionLimit > ...)
```

`getCacheClusters` eagerly called `DescribeReplicationGroups` in **every region** — a second paginated API round-trip per region — solely to pre-populate KMS-key IDs for the `kmsKey()` accessor (added in #7158 to avoid an N+1). Any query that never touches `kmsKey` still paid that cost, roughly doubling the latency of the whole multi-region `cacheClusters` walk.

## Fix

Make the KMS resolution lazy **without** reintroducing an N+1:

- Removed the eager per-region `DescribeReplicationGroups` from `getCacheClusters`.
- Added a mutex-guarded, per-region cache (`kmsKeyIdForReplicationGroup`) on the parent `aws.elasticache` resource. The first `kmsKey()` access in a region issues one `DescribeReplicationGroups` call for that region; subsequent clusters reuse the cached map.

Net effect:
- Queries that don't access `kmsKey` (the reported case) → **zero** replication-group calls.
- A full KMS scan → at most one `DescribeReplicationGroups` per region, same as before.

`CacheCluster` carries no KMS key of its own (only `ReplicationGroup` does), so behavior is unchanged.

## Verification

Built and installed the provider, then ran against live AWS using a temporary encrypted Redis replication group (one AWS-owned key, one customer-managed key), now torn down:

| Cluster | Encryption | `kmsKey` |
|---|---|---|
| `mql-test-5115-001` | AWS-owned (`sse-elasticache`) | `null` ✓ |
| `mql-test-5115-cmk-001` | customer-managed key | resolves to the CMK ARN ✓ |

- `make providers/build/aws` — builds, `aws.permissions.json` unchanged (911)
- `gofmt` clean, `golangci-lint` 0 issues
- Generated `aws.lr.go` diff limited to embedding the new internal struct; no `.lr.versions` changes

Fixes #5115

### CODE REVIEW

File: providers/aws/resources/aws_elasticache.go [hunk 1]
```diff
@@ -82,41 +83,58 @@ func (a *mqlAwsElasticache) getCacheClusters(conn *connection.AwsConnection) []*
...
+			for _, rg := range page.ReplicationGroups {
+				if rg.ReplicationGroupId != nil {
+					regionMap[*rg.ReplicationGroupId] = rg.KmsKeyId
 				}
 			}
-
-			return jobpool.JobResult(res), nil
 		}
-		tasks = append(tasks, jobpool.NewJob(f))
+		a.rgKmsByRegion[region] = regionMap
```
[the remark this answers is not in the document]
Comment (tas50) [reply providers/aws/resources/aws_elasticache.go]: Good catch — fixed in 4279c51b4. Replaced the single mutex with a `sync.Map` of per-region `sync.Once` entries (`rgKmsRegion`). Each region's `DescribeReplicationGroups` still runs at most once, but different regions now resolve concurrently, so we keep both wins: zero calls when `kmsKey` is never queried, and parallel per-region fetches when it is.

Confirmed live against the account with CMK-encrypted Redis replication groups in two regions (us-east-1 + us-west-1) — both `kmsKey` accessors resolved to the correct regional CMK ARN.

File: providers/aws/resources/aws_elasticache.go [hunk 2]
```diff
@@ -82,41 +83,66 @@ func (a *mqlAwsElasticache) getCacheClusters(conn *connection.AwsConnection) []*
...
+				if rg.ReplicationGroupId != nil {
+					keys[*rg.ReplicationGroupId] = rg.KmsKeyId
 				}
 			}
-
-			return jobpool.JobResult(res), nil
 		}
-		tasks = append(tasks, jobpool.NewJob(f))
+		entry.keys = keys
+	})
```
[the remark this answers is not in the document]
Comment (tas50) [reply providers/aws/resources/aws_elasticache.go]: Fixed in fd9877b02. Replaced the per-region `sync.Once` with a per-region mutex + `fetched` bool: a transient (non access-denied) `DescribeReplicationGroups` error now leaves the region unfetched, so a later `kmsKey()` access retries instead of returning the cached error forever. Access-denied / service-not-available are still treated as resolved-empty and cached once. Per-region keying is preserved, so the parallelism and lazy-skip wins are intact.

Also added a unit test for the `kmsKey()` null-state early-return (cluster with no replication group), mirroring `TestRdsKmsKey`. The populated/retry/concurrency paths in `kmsKeyIdForReplicationGroup` need a live elasticache client — there's no SDK-client mock pattern in the AWS provider, so those stay covered by interactive testing (verified live across two regions).
